lib/segment/analytics/request.rb

```diff
@@ -39,9 +40,14 @@ def post(write_key, batch)
         status, error = nil, nil
         remaining_retries = @retries
         backoff = @backoff
-        headers = { 'Content-Type' => 'application/json', 'accept' => 'application/json' }
+        # pass the API key also in the header, as expected by AWS API gateway
+        headers = { 'Content-Type' => 'application/json',
+                    'accept' => 'application/json',
+                    'x-api-key' => write_key }
```

Comment:
When using API Gateway (as in Polku) we need to pass the API key in the header. This should make the SDK work both with APIs that use basic HTTP authentication and API key-based authentication in AWS  API Gateway.